File: src/main/scala/com/github/dwendelen/platformd/rest/domain/budget/ExpensePost.java
package com.github.dwendelen.platformd.rest.domain.budget;

import com.datastax.driver.mapping.annotations.Column;
import com.datastax.driver.mapping.annotations.PartitionKey;
import com.datastax.driver.mapping.annotations.Table;

@Table(name = "expense_post")
public class ExpensePost {
    @PartitionKey
    @Column(name = "uuid")
    private String uuid;
    @Column(name = "name")
    private String name;

    public String getUuid() {
        return uuid;
    }

    public ExpensePost setUuid(String uuid) {
        this.uuid = uuid;
        return this;
    }

    public String getName() {
        return name;
    }

    public ExpensePost setName(String name) {
        this.name = name;
        return this;
    }
}

File: application/src/main/scala/com/github/dwendelen/platformd/rest/domain/budget/ExpensePost.scala
package com.github.dwendelen.platformd.rest.domain.budget

import java.util.UUID

import com.datastax.driver.mapping.annotations.{ClusteringColumn, Column, PartitionKey, Table}

import scala.annotation.meta.field
import scala.beans.BeanProperty

@Table(name = "expense_post") class ExpensePost(
    @(PartitionKey @field)
    @(Column @field)(name = "user_id")
    @BeanProperty
    var userId: UUID,
    @(ClusteringColumn @field)
    @(Column @field)(name = "expense_post_id")
    @BeanProperty
    var uuid: UUID,
    @(Column @field)(name = "name")
    @BeanProperty
    var name: String
){
    def this() = this(null, null, null)
}